cmd/signozschemamigrator/schema_migrator/column_operations.go

```diff
@@ -0,0 +1,446 @@
+package schemamigrator
+
+import (
+	"strings"
+)
+
+// AlterTableAddColumn is used to add a column to a table.
+// It is used to represent the ALTER TABLE ADD COLUMN statement in the SQL.
+type AlterTableAddColumn struct {
+	cluster string
+
+	Database string
+	Table    string
+	Column   Column
+	// Should be used carefully, this is to be used when the column is to be added after a specific column
+	// If not specified, the column will be added at the end
+	After *Column
+}
+
+// OnCluster is used to specify the cluster on which the operation should be performed.
+// This is useful when the operation is to be performed on a cluster setup.
+func (a AlterTableAddColumn) OnCluster(cluster string) Operation {
+	a.cluster = cluster
+	return &a
+}
+
+func (a AlterTableAddColumn) WithReplication() Operation {
+	// no-op
+	return &a
+}
+
+func (a AlterTableAddColumn) ShouldWaitForDistributionQueue() (bool, string, string) {
+	return false, a.Database, a.Table
+}
+
+func (a AlterTableAddColumn) IsMutation() bool {
+	// Adding a column is not a mutation. It simply updates the metadata of the table.
+	return false
+}
+
+func (a AlterTableAddColumn) IsIdempotent() bool {
+	// Adding a column is idempotent. It will not change the table if the column already exists.
+	return true
+}
+
+func (a AlterTableAddColumn) IsLightweight() bool {
+	// Adding a column is lightweight. It simply updates the metadata of the table.
+	return true
+}
+
+func (a AlterTableAddColumn) ToSQL() string {
+	var sql strings.Builder
+	sql.WriteString("ALTER TABLE ")
+	sql.WriteString(a.Database)
+	sql.WriteString(".")
+	sql.WriteString(a.Table)
+	if a.cluster != "" {
+		sql.WriteString(" ON CLUSTER ")
+		sql.WriteString(a.cluster)
+	}
+	sql.WriteString(" ADD COLUMN IF NOT EXISTS ")
+	sql.WriteString(a.Column.ToSQL())
+	if a.After != nil {
+		sql.WriteString(" AFTER ")
+		sql.WriteString(a.After.Name)
+	}
+	return sql.String()
```

Comment:
codec for the column?

Reply:
It is already part of `sql.WriteString(a.Column.ToSQL())`

Reply:
Ahh, can you modify a add column test case and add it ?

Reply:
Added in https://github.com/SigNoz/signoz-otel-collector/pull/390/commits/0c83188bcfa5d0322a5b9f5dfb34314378d7ad18